lib/containers/common.pm

```diff
@@ -241,16 +227,17 @@ sub test_container_image {
         assert_script_run("buildah run $container $smoketest");
     } else {
         # Pull the image if necessary
-        if (script_run("$runtime image inspect --format='{{.RepoTags}}' $image | grep '$image'") != 0) {
+        if (script_run("$runtime->{runtime} image inspect --format='{{.RepoTags}}' $image | grep '$image'") != 0) {
             # At least on publiccloud, this image pull can take long and occasinally fails due to network issues
-            script_retry("$runtime pull $image", retry => 3, timeout => 420);
-            assert_script_run("$runtime image inspect --format='{{.RepoTags}}' $image | grep '$image'");
+            $runtime->pull($image, timeout => 420);
+            $runtime->confirm_image_in_host_registry($image);
         }
-        assert_script_run("$runtime container create --name testing $image $smoketest");
-        assert_script_run("$runtime container start 'testing'");
-        assert_script_run("$runtime wait 'testing'", 90);
-        assert_script_run("$runtime container logs 'testing' | tee '$logfile'");
-        assert_script_run("$runtime container rm 'testing'");
+        $runtime->create_container($image, 'testing', $smoketest);
+        $runtime->start_container('testing');
+        $runtime->halt_container('testing');
+        my $logs = $runtime->get_container_logs('testing');
+        assert_script_run "echo '$logs' | tee '$logfile'";
```

Comment:
This is definitely not good. Why don't you make `$runtime->print_container_logs('testing');` instead?

Reply:
As you can see i worked around `assert_script_run("$runtime container logs 'testing' | tee '$logfile'");`
Again to keep it minimal i avoided some things. improvements are to be done afterwards. What i am looking now is 
1. does it break anything
2. do i miss something 
3. is the approach and the design acceptable

Reply:
I still don't understand.